src/panels/config/zha/zha-config-dashboard.ts

```diff
@@ -0,0 +1,214 @@
+import memoizeOne from "memoize-one";
+import { DataTableColumnContainer } from "../../../components/data-table/ha-data-table";
+import { showZHADeviceInfoDialog } from "../../../dialogs/zha-device-info-dialog/show-dialog-zha-device-info";
+
+export interface DeviceRowData extends ZHADevice {
+  device?: DeviceRowData;
+}
+
+@customElement("zha-config-dashboard")
+class ZHAConfigDashboard extends LitElement {
+  @property() public hass!: HomeAssistant;
+  @property() public route!: Route;
+  @property() public narrow!: boolean;
+  @property() public isWide!: boolean;
+  @property() private _devices: ZHADevice[] = [];
+  private pages: string[] = ["add", "groups"];
+  private _firstUpdatedCalled: boolean = false;
+
+  private _memoizeDevices = memoizeOne((devices: ZHADevice[]) => {
+    let outputDevices: DeviceRowData[] = devices;
+
+    outputDevices = outputDevices.map((device) => {
+      return {
+        ...device,
+        name: device.user_given_name ? device.user_given_name : device.name,
+        nwk: formatAsPaddedHex(device.nwk),
+        ieee: device.ieee,
+        id: device.ieee,
+      };
+    });
+
+    return outputDevices;
+  });
+
+  private _columns = memoizeOne(
+    (narrow: boolean): DataTableColumnContainer =>
+      narrow
+        ? {
+            name: {
+              title: "Devices",
+              sortable: true,
+              filterable: true,
+              direction: "asc",
+              template: (name) => html`
+                <div
+                  @click=${this._handleDeviceClicked}
+                  style="cursor: pointer;"
+                >
+                  ${name}
+                </div>
+              `,
+            },
+            id: {
+              title: "",
+              sortable: false,
+              filterable: false,
+              template: (id) => html`
+                <a href=${`/config/zha/device/${id}`}>
+                  <ha-icon-next
+                    style="color: var(--primary-text-color);"
+                  ></ha-icon-next>
+                </a>
+              `,
+            },
+          }
+        : {
+            name: {
+              title: "Name",
+              sortable: true,
+              filterable: true,
+              direction: "asc",
+              template: (name) => html`
+                <div
+                  @click=${this._handleDeviceClicked}
+                  style="cursor: pointer;"
+                >
+                  ${name}
+                </div>
+              `,
+            },
+            nwk: {
+              title: "Nwk",
+              sortable: true,
+              filterable: true,
+            },
+            ieee: {
+              title: "IEEE",
+              sortable: true,
+              filterable: true,
+            },
+            id: {
+              title: "",
+              sortable: false,
+              filterable: false,
+              template: (id) => html`
+                <a href=${`/config/zha/device/${id}`}>
```

Comment:
Why are there 2 different actions, one for a dialog and one for a device page? Can't those be combined?

Reply:
you can see it in the video above. Is there a reason it can't be like this?

Reply:
The video doesn't have a pause button so hard to see ;-)

I think it just makes it complicated, to have 2 actions on 1 row and to have 2 screens with device options/info. I would just combine it all in the device page.

Reply:
Done